OpenRA.Mods.Common/Activities/Air/HeliReturnToBase.cs

```diff
@@ -44,16 +44,33 @@ public override Activity Tick(Actor self)
 			{
 				var rearmBuildings = heli.Info.RearmBuildings;
 				var nearestHpad = self.World.ActorsHavingTrait<Reservable>()
-									.Where(a => a.Owner == self.Owner && rearmBuildings.Contains(a.Info.Name))
-									.ClosestTo(self);
+					.Where(a => a.Owner == self.Owner && rearmBuildings.Contains(a.Info.Name))
+					.ClosestTo(self);
 
 				if (nearestHpad == null)
 					return ActivityUtils.SequenceActivities(new Turn(self, initialFacing), new HeliLand(self, true), NextActivity);
 				else
-					return ActivityUtils.SequenceActivities(new HeliFly(self, Target.FromActor(nearestHpad)));
+				{
+					var distanceFromHelipad = (nearestHpad.CenterPosition - self.CenterPosition).HorizontalLength;
+					var distanceLength = heli.Info.WaitDistanceFromResupplyBase.Length;
+
+					// If no pad is available, move near one and wait
+					if (distanceFromHelipad > distanceLength)
+					{
+						var randomXPosition = self.World.SharedRandom.Next(-distanceLength, distanceLength);
+						var randomYPosition = self.World.SharedRandom.Next(-distanceLength, distanceLength);
+
+						var target = Target.FromPos(nearestHpad.CenterPosition + new WVec(randomXPosition, randomYPosition, 0));
+
+						return ActivityUtils.SequenceActivities(new HeliFly(self, target, WDist.Zero, heli.Info.WaitDistanceFromResupplyBase), this);
+					}
+
+					return this;
+				}
 			}
 
 			heli.MakeReservation(dest);
+			heli.DisableRepulsing();
```

Comment:
This looks dangerous.  What do we gain by manually disabling and enabling repulsation?